include/ideep/attributes.hpp

```diff
@@ -288,6 +303,41 @@ struct attr_t : public dnnl::primitive_attr {
     return true;
   }
 
+  attr_t& operator=(const attr_t& rhs) {
+    if (this == &rhs) {
+      return *this;
+    }
+    dnnl_primitive_attr_t result;
+    error::wrap_c_api(
+        dnnl_primitive_attr_clone(&result, rhs.get()),
+        "could not clone primitive attributes");
+    this->reset(result);
+    return *this;
+  }
+
+  bool operator==(const attr_t& rhs) const {
+    auto l_po = get_post_ops();
+    auto r_po = rhs.get_post_ops();
+    if (l_po.len() != r_po.len() ||
+        get_output_scales() != rhs.get_output_scales()) {
+      return false;
+    }
+    for (auto index = 0; index < l_po.len(); index++) {
+      kind l_akind, r_akind;
+      algorithm l_alg, r_alg;
+      float l_scale = 1.0, l_alpha = 1.0, l_beta = 0.0;
+      float r_scale = 1.0, r_alpha = 1.0, r_beta = 0.0;
+      std::tie(l_akind, l_scale, l_alpha, l_beta, l_alg) = get_params(index);
+      std::tie(r_akind, r_scale, r_alpha, r_beta, r_alg) =
+          rhs.get_params(index);
+      if (l_akind != r_akind || l_alg != r_alg || l_scale != r_scale ||
+          l_alpha != r_alpha || l_beta != r_beta) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
Are you compare and copy fpmath_mode for attr_t?

Reply:
This implementation is from IPEX. Looks like it does not compare everything. I will sync with the code owner and update it in the new PR if needed.

Reply:
I have updated the implementation of operator == to consider more aspects when comparing.